engine/runtime/src/main/java/org/enso/interpreter/runtime/state/State.java

```diff
@@ -0,0 +1,22 @@
+package org.enso.interpreter.runtime.state;
+
+import com.oracle.truffle.api.object.DynamicObject;
+import com.oracle.truffle.api.object.Shape;
+
+public record State(Container container, IOPermissions ioPermissions) {
```

Comment:
Probably you also want a `private` constructor with a factory method to get the default `State`.